Sources/Basics/EnvironmentVariables.swift

```diff
@@ -64,7 +64,7 @@ extension EnvironmentVariables {
 // rdar://107029374
 extension EnvironmentVariables {
     // internal for testing
-    internal static let nonCachableKeys: Set<String> = [
+    static let nonCachableKeys: Set<String> = [
```

Comment:
Would it make sense to have an list that we know *should* cause rebuilds rather than the other way around?

Reply:
Not until we support something like package traits to customise packages through predefined configuration values. Implementing that requires major changes to package graph resolution and manifest evaluation code.

Right now the only way for users to configure packages at build time is to pass an environment variable to `swift build` and read that variable in `Package.swift`. We can't know upfront which environment variables will be used for customisation by an arbitrary package, and changes in such variables have to cause rebuilds for this customisations to work in the first place.